common/src/main/java/io/netty/util/concurrent/SingleThreadEventExecutor.java

```diff
@@ -419,7 +419,8 @@ protected void wakeup(boolean inEventLoop) {
 
     @Override
     public boolean inEventLoop(Thread thread) {
-        return thread == this.thread;
+        return thread == this.thread ||
+                executor instanceof EventExecutor && ((EventExecutor) executor).inEventLoop(thread);
```

Comment:
Understood. I'm fine with tracking this via and issue and a comment in the code that this is quirky and then resolving in a separate PR.